Antidote/ActiveCallViewController.m

```diff
@@ -153,17 +159,53 @@ - (void)installConstraints
 
 }
 
-#pragma mark - Private
+#pragma mark - Public
 
-- (void)didUpdateCall
+- (void)setCallDuration:(NSTimeInterval)callDuration
 {
-    [super didUpdateCall];
+    _callDuration = callDuration;
     [self updateTimerLabel];
 }
 
+- (void)setMicSelected:(BOOL)micSelected
+{
+    self.microphoneButton.selected = micSelected;
+}
+
+- (BOOL)micSelected
+{
+    return self.microphoneButton.selected;
+}
+
+- (void)setSpeakerSelected:(BOOL)speakerSelected
+{
+    self.speakerButton.selected = speakerSelected;
+}
+
+- (BOOL)speakerSelected
+{
+    return self.speakerButton.selected;
+}
+
+- (void)setShowIncomingCallView:(BOOL)showIncomingCallView
+{
+    if (showIncomingCallView) {
+        [self setupIncomingCallView];
```

Comment:
You cannot know what will happen in a month or in a year. CallsManager can be dead at this point. If you can easily save future yourself (or someone else) from bugs, why not to do that? :-)

Just add check at beginning of method
```
if (showIncomingCallView == _showIncomingCallView) {
    return;
}
```